project-base/storefront/gtm/types/events.ts

```diff
@@ -6,6 +6,7 @@ import {
     GtmPageInfoType,
```

Comment:
I do not see the user object added to page_ready, ec.create_order, ec.payment, ec.login, and ec.registration as mentioned in AC.

Reply:
`ec.login` & `ec.registration` events were added and `onGthUserEntryEventHandler` is sent in `useAfterUserEntry` hook after the user is correctly redirected after login/registration. The correct event type is retrieved from the persist store.

As for the rest, there were some misspelt/wrong events (page_ready => page_view, ec.payment being actually ec.create_order). For the `page_view` and `ec.create_order`, there already was a user so I have extended the `GtmUserInfoType` type and `getGtmUserInfo` function to include information about the user loginType and loginUserId.